collections_hierarchy/mixins.py

```diff
@@ -1,87 +1,103 @@
 class ComparableMixin(object):
     def __eq__(self, other):
-        pass
+        for x, y in zip(self, other):
+            if not x == y:
+                pass
+            else:
+                return self == other
 
     def __ne__(self, other):
-        return not self == other
+        for x, y in zip(self, other):
+            if x == y:
+                pass
+            else:
+                return not self == other
 
     def __lt__(self, other):
-        pass
+        return self < other
 
     def __gt__(self, other):
-        pass
+        return self > other
 
     def __le__(self, other):
-        pass
+        return self <= other
 
     def __ge__(self, other):
-        pass
+        return self >= other
 
 
 class SequenceMixin(object):
     def __iter__(self):
-        pass
+        self.n = 0
+        self.elements = self.get_elements()
+        return(self)
 
     def __next__(self):
-        pass
+        while self.n < len(self.elements):
+            element = self.elements[self.n]
+            self.n += 1
+            return element
 
+        raise StopIteration
+            
     next = __next__
 
     def __len__(self):
-        pass
+        return len(self)
 
     def __getitem__(self, key):
-        pass
+        return self[key]
 
     def __setitem__(self, key, value):
-        pass
+        self[key] = value
 
     def __delitem__(self, key):
-        pass
+        del self[key]
 
     def __contains__(self, item):
-        pass
+        return item in self.elements
 
 
 class RepresentableMixin(object):
     def __repr__(self):
-        pass
+        return str(self.elements)
 
     def __str__(self):
-        pass
+        return str(self.elements)
 
 
 class ConstructibleMixin(object):
     DATA_ATTR_NAME = 'data'
 
     def __init__(self, initial=None):
-        pass
-
+        self.initial = initial or self.DATA_DEFAULT_INITIAL
+        setattr(self, self.DATA_ATTR_NAME, self.initial)
+        
 
 class OperableMixin(object):
     def __add__(self, other):
-        pass
+        return self.elements + other.elements
```

Comment:
We did this at first, too.  The only problem is that what it returns should be a `List`, but it's actually just a `list` (and it causes a test to fail).  You can do some maneuvering with `append` to define this behavior in such a way that it returns the right type.